apps/prairielearn/python/question_phases.py

```diff
@@ -150,7 +168,15 @@ def process_element(
             # Restore the tail text.
             element.tail = temp_tail
 
-            check_data(old_data, data, phase)
+            check_data(original_data, data, phase)
+
+            # Clean up changes to `data` and `original_data` for the next iteration.
+            data.pop("extensions", None)
```

Comment:
This deletion logic can be factored out into a helper function, but not a huge deal since this is only a couple of lines.

Reply:
I did that refactor in 33fa31e; I also moved the logic to _add_ to `data`/`original_data` to a helper function. Does that look good?

Reply:
Just approved, changes make the code much easier to follow imo.